lib/posix/pthread.c

```diff
@@ -191,7 +190,7 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *_attr,
 	rv = pthread_mutex_init(&thread->state_lock, NULL);
 	if (rv != 0) {
 		key = k_spin_lock(&pthread_pool_lock);
-		thread->state = PTHREAD_EXITED;
+		thread->state = PTHREAD_TERMINATED;
```

Comment:
Yes, this looks like there was just a bug here -- there's no way you'd be able to join this thread as it was never returned, right?

Reply:
Previously I added a change that allowed EXITED as well. But yeah, there was no previous documentation. I would definitely like to add some, but that is not as high priority